Return a FlexVersion object instead of a String.

kostasniktas opened the pull request:
-This will allow a user to extract information from the version w/o needing to parse
-Gradle is smart enough to take an Object for a version so this will work as it calls toString
-Anything expecting an actual String will fail and need to change to have a toString()

Commit:
Return a FlexVersion object instead of a String.

-This will allow a user to extract information from the version w/o needing to parse
-Gradle is smart enough to take an Object for a version so this will work as it calls toString
-Anything expecting an actual String will fail and need to change to have a toString()

kostasniktas commented:
@EXBreder 
@atrocities

imawuman commented on src/main/groovy/com/palantir/gradle/versions/flexversioning/FlexVersion.groovy:
```diff
@@ -0,0 +1,35 @@
+    boolean dirty = false;
+    boolean tag = false;
+
+    public FlexVersion (String domain, String commitCount, String gitHash, boolean dirty) {
```
this class should probably have a build() method that the toString() method just calls rather than storing the fullVersion as a variable and computing it outside the object.

example use case would be if i wanted to get the flex version object, use its parts for almanac publishing, then assign the full version string as metadata. I want to do all of those steps using the same flex version object.

Commit:
Move the logic for building a version string from pieces into object

-The actual picking up the domain is still up to the plugin
-The FlexVersion object itself builds the string from all the pieces

kostasniktas commented:
As per @EXBreder  and Richard's comments, I moved the string building logic into the object itself.  It's always going to be an integer, I just had kept the string that it was going.  Changed to be an int.

Commit:
FlexVersion object. Clean up a few more things
